be/src/exec/vectorized/aggregate/distinct_blocking_node.cpp

```diff
@@ -148,7 +149,14 @@ std::vector<std::shared_ptr<pipeline::OperatorFactory> > DistinctBlockingNode::d
     // Initialize OperatorFactory's fields involving runtime filters.
     this->init_runtime_filter_for_operator(source_operator.get(), context, rc_rf_probe_collector);
 
-    operators_with_sink = context->maybe_interpolate_local_shuffle_exchange(operators_with_sink, partition_expr_ctxs);
+    // The san operator followed by an aggregate operator indicates the aggregating key is the same as
+    // the bucket key of this table, so we needn't local shuffle for this case.
+    auto* source_op = operators_with_sink[0].get();
+    if (operators_with_sink.size() > 1 || typeid(*source_op) != typeid(pipeline::ScanOperatorFactory)) {
```

Comment:
typid cannot judge derive class or base class, you could replace it with dynamic_cast<PointerType>